internal/utils/serializer.go

```diff
@@ -14,6 +14,6 @@ func ContextToMap(ctx ffcontext.Context) map[string]interface{} {
 		userCopy[key] = value
 	}
 	userCopy["anonymous"] = ctx.IsAnonymous()
-	userCopy["key"] = ctx.GetKey()
+	userCopy["key"] = key
```

Comment:
As mentioned in the issue, I'd prefer not to do this.
`key` and `targetingKey` are special fields in the evaluation context and we should keep them like that.

For this feature, the important part is to be sure that we split the users based on another field it should not have an impact on the way to write rules.

This part will change how you write your rules and may lead to misleading configuration of your flags.

Let's say you have this flag
```yaml
my-flag:
  variation:
     enabled: true
     disabled: false
  defaultRule:
     percentage:
       enabled: 10
       disabled: 90
  bucketingKey: teamId
```
and this evaluation context:
```json
{
  "targetingKey": "f1f0d804-f23f-4c93-a935-6e04ce2cf696",
  "teamId": 567,
  "email": "contact@gofeatureflag.org"
}
```

If you have to add a rule inside the flag based on the `teamId` you can go with
```yaml
# ...
  targeting:
     - query: teamId eq "567"
       variation: enabled
# ...
```
I don't see the adding value to also put `teamId` value as `key`.